pkg/controller/service/controller.go

```diff
@@ -348,6 +355,7 @@ func (s *Controller) syncLoadBalancerIfNeeded(service *v1.Service, key string) (
 				// functionality is implemented by a different controller.  In this case, we
 				// return immediately without doing anything.
 				klog.V(4).Infof("LoadBalancer for service %s implemented by a different controller %s, Ignoring error", key, s.cloud.ProviderName())
+				s.removeFinalizer(service)
```

Comment:
Check for error?

Reply:
Would leaving the default finalizer attached cause problem?

Reply:
The finalizer removal was added so that the check in line 337 would work for services that are managed by a different controller. The condition is - load-balancer-cleanup finalizer should not be present AND an unrecognized finalizer should be present.

Your recommendation to check explicitly for the v2 finalizer in gce provider code is good, we can move the check there and leave the load-balancer-cleanup finalizer intact in all cases.